src/components/App/App.js

```diff
@@ -50,22 +97,72 @@ function App(props) {
       [modalId]: false, [newModalId]: true});
   }
 
+  async function registerUser(name, email, password) {
+    return userApi.addUser({ name, email, password })
+      .then((res) => {
+        setToken(res.token);
+        auth(res.token);
+        handleModalClose("signup");
+      });
+  }
+
+  async function signIn(email, password) {
+    return userApi.signIn({ email, password })
+      .then((res) => {
+        setToken(res.token);
+        auth(res.token);
+        handleModalClose("login");
+      })
+      .then(() => {
+        if (keyword) {
+          getNews(keyword);
+          getSavedArticles();
+        }
+      });
+  }
+
+  function logOut() {
+    removeToken();
+    setIsLoggedIn(false);
+    setArticles(null);
+  }
+
+  function auth(token) {
+    userApi.auth(token)
+      .then((res) => {
+        userApi.setTokenHeader(token);
+        setCurrentUser(res.data);
+        setIsLoggedIn(true);
+        getSavedArticles();
+      })
+      .catch((err) => { console.log(err) });
+  }
+
   //#endregion
 
   
   //#region Variables setup
 
-  const [isLoggedIn, setIsLoggedIn] = useState(true);
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
   const [news, setNews] = useState(null);
   const [isSearching, setIsSearching] = useState(false);
   const [modalsActivity, setModalsActivity] = useState({
     "signup": false,
     "login": false,
     "success": false,
-  });
 
+  });
+  const [currentUser, setCurrentUser] = useState({});
+  const [articles, setArticles] = useState(null);
+  const [keyword, setKeyword] = useState(null);
   const [isOnMobile, setIsOnMobile] = useState(window.innerWidth < 600);
+
   useEffect(() => {
+    const token = getToken();
+    if (token) {
+      auth(token);
+    }
+    
     window.addEventListener("resize", () => {
       setIsOnMobile(window.innerWidth < 600);
     });
```

Comment:
```js

    window.addEventListener("resize", () => {
      setIsOnMobile(window.innerWidth < 600);
    });
    return () => {
      window.removeEventListener("resize", () => {
        setIsOnMobile(window.innerWidth < 600);
      })
      }
  }, []);
```

You are not able to remove any listener in such a way.

More info here: https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener